src/ingameWindows/iwMilitaryBuilding.cpp

```diff
@@ -95,9 +95,9 @@ void iwMilitaryBuilding::Msg_PaintAfter()
     DrawRectangle(GetX() + width_ / 2 - 22 * TROOPS_COUNT[building->nation][building->size] / 2, GetY() + 98 , 22 * TROOPS_COUNT[building->nation][building->size], 24, 0x96000000);
 
     // Sammeln aus der Rausgeh-Liste und denen, die wirklich noch drinne sind
-    std::multiset<unsigned> soldiers;
+    std::multiset<nofSoldier*, ComparatorSoldiersByRank<true>> soldiers;
```

Comment:
This should even be a `const nofSoldier*` so one does not even try to modify them